beluga_ros/src/amcl.cpp

```diff
@@ -50,6 +50,26 @@ void Amcl::update_map(beluga_ros::OccupancyGrid map) {
   std::visit([&](auto& sensor_model) { sensor_model.update_map(std::move(map)); }, sensor_model_);
 }
 
+void Amcl::process_buffered_odometry_until(std::deque<OdometryMotion>& buffer, const tf2::TimePoint& until) {
+  while (!buffer.empty()) {
+    const auto& [odom_time, odom_pose] = buffer.front();
+    if (odom_time >= until)
+      break;
+    update(odom_pose);
+    buffer.pop_front();
+  }
+}
```

Comment:
it sounds to me like this belongs in beluga_amcl.cpp, justl ike the OdometryMotion datatype, since implementing the odometry message queue is that file's concern.

Reply:
My only comment here, and it is as meta as https://github.com/Ekumen-OS/beluga/pull/492#discussion_r2285113313 was, is that processing a longer window of odometry measurements is something that could be pushed to the core motion model. This is like conditioning the proposal distribution on a control sequence that is longer than two elements.

Really, it is a non-issue here. Keep it simple. It can be explored in follow-up work.